File: app/src/main/java/com/examples/android/androidtrainingbzu/DialogsActivity.java
package com.examples.android.androidtrainingbzu;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.graphics.Color;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.DatePicker;
import android.widget.TextView;
import android.widget.Toast;
import android.widget.ToggleButton;

import com.examples.android.androidtrainingbzu.Fragments.TimePickerFragment;

import java.util.Calendar;

public class DialogsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dialogs);
    }
    @Override
    public void onBackPressed() {
        AlertDialog.Builder alertDialog=new AlertDialog.Builder(this);
        alertDialog.setTitle("Exit? -Dialog Title-");
        alertDialog.setMessage("Exit the current activity? -Dialog Message-");
        alertDialog.setPositiveButton("Yes", new DialogInterface.OnClickListener(){
            @Override
            public void onClick(DialogInterface dialog, int which) {
                finish();
            }
        });
        alertDialog.setNegativeButton("No", new DialogInterface.OnClickListener(){
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        alertDialog.setCancelable(false);
        alertDialog.show();
    }

    public void pickDate(View view) {
        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);
        Dialog datePickerDialog=new DatePickerDialog(this, new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
                processDatePickerResult(year, month, dayOfMonth);
            }
        }, year, month, day);

        datePickerDialog.show();
    }

    public void pickTime(View view) {
        DialogFragment newFragment = new TimePickerFragment();
        newFragment.show(getSupportFragmentManager(), getString(R.string.time_picker));
    }
    public void processDatePickerResult(int year, int month, int day) {
        // The month integer returned by the date picker starts counting at 0
        // for January, so you need to add 1 to show months starting at 1.
        String month_string = Integer.toString(month + 1);
        String day_string = Integer.toString(day);
        String year_string = Integer.toString(year);
        // Assign the concatenated strings to dateMessage.
        String dateMessage = (month_string + "/" + day_string + "/" + year_string);
        ((TextView)findViewById(R.id.txt_date)).setText("Date : "+dateMessage);
    }

    public void processTimePickerResult(int hourOfDay, int minute) {
        // Convert time elements into strings.
        String hour_string = Integer.toString(hourOfDay);
        String minute_string = Integer.toString(minute);
        // Assign the concatenated strings to timeMessage.
        String timeMessage = (hour_string + ":" + minute_string);
        ((TextView)findViewById(R.id.txt_time)).setText("Time : "+timeMessage);
    }
    Snackbar snackbar=null;
    public void showSnackbar(View view){
        snackbar = Snackbar
                .make(view, "This is a Snackbar, show Toast?", Snackbar.LENGTH_LONG)
                .setAction("Yes", new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        Toast.makeText(getApplicationContext(), "Snackbar action button clicked",Toast.LENGTH_LONG).show();
                    }
                });
        snackbar.setActionTextColor(Color.RED);  //getResources().getColor(R.color.green));
        View sbView = snackbar.getView();
        TextView textView = (TextView) sbView.findViewById(android.support.design.R.id.snackbar_text);
        textView.setTextColor(Color.YELLOW);
        snackbar.show();
    }
}

File: app/src/main/java/com/examples/android/androidtrainingbzu/DialogsActivity.kt
package com.examples.android.androidtrainingbzu

import android.app.DatePickerDialog
import android.app.Dialog
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import com.examples.android.androidtrainingbzu.Fragments.TimePickerFragment
import com.google.android.material.snackbar.Snackbar
import java.util.*

class DialogsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_dialogs)
    }

    override fun onBackPressed() {
        val alertDialog = AlertDialog.Builder(this)
        alertDialog.setTitle("Exit? -Dialog Title-")
        alertDialog.setMessage("Exit the current activity? -Dialog Message-")
        alertDialog.setPositiveButton("Yes") { dialog, which -> finish() }
        alertDialog.setNegativeButton("No") { dialog, which -> dialog.dismiss() }
        alertDialog.setCancelable(false)
        alertDialog.show()
    }

    fun pickDate(view: View?) {
        val c = Calendar.getInstance()
        val year = c[Calendar.YEAR]
        val month = c[Calendar.MONTH]
        val day = c[Calendar.DAY_OF_MONTH]
        val datePickerDialog: Dialog = DatePickerDialog(this,
            { view, year, month, dayOfMonth -> processDatePickerResult(year, month, dayOfMonth) },
            year,
            month,
            day)
        datePickerDialog.show()
    }

    fun pickTime(view: View?) {
        val newFragment: DialogFragment = TimePickerFragment()
        newFragment.show(supportFragmentManager, getString(R.string.time_picker))
    }

    fun processDatePickerResult(year: Int, month: Int, day: Int) {
        // The month integer returned by the date picker starts counting at 0
        // for January, so you need to add 1 to show months starting at 1.
        val month_string = Integer.toString(month + 1)
        val day_string = Integer.toString(day)
        val year_string = Integer.toString(year)
        // Assign the concatenated strings to dateMessage.
        val dateMessage = "$month_string/$day_string/$year_string"
        (findViewById<View>(R.id.txt_date) as TextView).text = "Date : $dateMessage"
    }

    fun processTimePickerResult(hourOfDay: Int, minute: Int) {
        // Convert time elements into strings.
        val hour_string = Integer.toString(hourOfDay)
        val minute_string = Integer.toString(minute)
        // Assign the concatenated strings to timeMessage.
        val timeMessage = "$hour_string:$minute_string"
        (findViewById<View>(R.id.txt_time) as TextView).text = "Time : $timeMessage"
    }

    var snackbar: Snackbar? = null
    fun showSnackbar(view: View?) {
        snackbar = Snackbar
            .make(view!!, "This is a Snackbar, show Toast?", Snackbar.LENGTH_LONG)
            .setAction("Yes") {
                Toast.makeText(applicationContext,
                    "Snackbar action button clicked",
                    Toast.LENGTH_LONG).show()
            }
        snackbar!!.setActionTextColor(Color.RED) //getResources().getColor(R.color.green));
        val sbView = snackbar!!.view
        val textView =
            sbView.findViewById<View>(com.google.android.material.R.id.snackbar_text) as TextView
        textView.setTextColor(Color.YELLOW)
        snackbar!!.show()
    }
}